src/main/java/uk/gov/pay/adminusers/client/ledger/service/LedgerService.java

```diff
@@ -37,6 +41,25 @@ public Optional<LedgerTransaction> getTransaction(String id) {
         return getTransactionFromLedger(uri);
     }
 
+    public LedgerSearchTransactionsResponse searchTransactions(String gatewayAccountId, int limit) {
+        var uri = UriBuilder
+                .fromPath(ledgerUrl)
+                .path("/v1/transaction")
+                .queryParam("account_id", gatewayAccountId)
+                .queryParam("display_size", limit);
```

Comment:
Would we not want to add a query param for created date so we can query for transactions less than 7 years old? I'm not sure how `/v1/transaction` is implemented in ledger and how results are retrieved, but I think having a query param for created date would be more explicit.

Reply:
We don't need to query by created_date. We will search ledger for a transaction (Ledger returns the last transaction). Adminusers will check if the last transaction is 7 years old or not and archive service accordingly